### ISSUE TITLE
[CRASH] Unhandled 'error' event in OLCValidator

### ISSUE THREAD

**Issue by julianweise** [bptlab/gryphon#31]
Gryphon crashes after submitting a new configuration for service task. I tried to reference an attribute of an data object within the URL field of the webtask.

web service URL field content: `https://hpi.goodstag.com/api/tags/#BookTag.NFCID`

**Error log:**
```
gryphon_1   | events.js:167
gryphon_1   |       throw er; // Unhandled 'error' event
gryphon_1   |       ^
gryphon_1   |
gryphon_1   | TypeError: Cannot read property '0' of undefined
gryphon_1   |     at OLCValidator.<anonymous> (/var/gryphon/server_src/helpers/olcvalidator.js:64:75)
gryphon_1   |     at Array.forEach (<anonymous>)
gryphon_1   |     at OLCValidator.<anonymous> (/var/gryphon/server_src/helpers/olcvalidator.js:63:43)
gryphon_1   |     at Array.forEach (<anonymous>)
gryphon_1   |     at OLCValidator.validateDataObjectFlow (/var/gryphon/server_src/helpers/olcvalidator.js:48:41)
gryphon_1   |     at OLCValidator.validateEverything (/var/gryphon/server_src/helpers/olcvalidator.js:30:14)
gryphon_1   |     at GeneralValidator.validateWithSimpleValidator (/var/gryphon/server_src/helpers/validator.js:77:19)
gryphon_1   |     at GeneralValidator.<anonymous> (/var/gryphon/server_src/helpers/validator.js:67:18)
gryphon_1   |     at Array.forEach (<anonymous>)
gryphon_1   |     at GeneralValidator.validateEverything (/var/gryphon/server_src/helpers/validator.js:59:25)
gryphon_1   |     at /var/gryphon/server_src/routes/fragment.js:208:27
gryphon_1   |     at GeneralValidator.<anonymous> (/var/gryphon/server_src/helpers/validator.js:50:13)
gryphon_1   |     at /var/gryphon/node_modules/mongoose/lib/model.js:4208:16
gryphon_1   |     at (anonymous function).call (/var/gryphon/node_modules/mongoose/lib/query.js:3128:7)
gryphon_1   |     at process.nextTick (/var/gryphon/node_modules/mongoose/lib/query.js:2019:28)
gryphon_1   |     at process._tickCallback (internal/process/next_tick.js:61:11)
gryphon_1   | Emitted 'error' event at:
gryphon_1   |     at /var/gryphon/node_modules/mongoose/lib/model.js:4210:13
gryphon_1   |     at (anonymous function).call (/var/gryphon/node_modules/mongoose/lib/query.js:3128:7)
gryphon_1   |     at process.nextTick (/var/gryphon/node_modules/mongoose/lib/query.js:2019:28)
gryphon_1   |     at process._tickCallback (internal/process/next_tick.js:61:11)
```

**Comment by tommartensen** [comment]
This is difficult to solve, since the `#` is used in URIs to identify anchor links. We could probably solve this issue here (and in Chimera), but maybe there should be a better way to reference data objects!

**Comment by tommartensen** [comment]
@marhew So in your eyes, this was the user input that caused the crash? 
@julianweise Can you confirm Marcin from your memory?

**Comment by julianweise** [comment]
I had a deeper look into the issue. It seems like the issue always occurs if you name the activity before changing the name and filling out the fields of an service task.
What I did in order not to run in the above given exception anymore during modeling was:

1. Create a new activity (do not give it a name!)
2. Connect it
3. Change type to `ServiceTask`
4. Fill out the required fields
5. Name activity

**Comment by marhew** [comment]
This was the same error that was responsible for #32 it should also be fixed by 2bb1a199cd35ea2eda6c8c88dcdee73f870131b3
